### ISSUE TITLE
Multiple test blocks are running despite `.only`

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#632]
Issue in https://github.com/bahmutov/effective-e2e-testing-with-cypress/issues/1 code in https://github.com/bahmutov/effective-e2e-testing-with-cypress/tree/double-only

I think this is because one block has title that is substring of another block

### Current behavior:

```js
describe.only('main features', function () {
  it('first spec', function () {})
})

describe('main features with assertions', function () {
  it('second spec', function () {})
})
```
Runs both blocks despite `.only`

<img width="1273" alt="screen shot 2017-09-11 at 4 38 33 pm" src="https://user-images.githubusercontent.com/2212006/30296253-8bb16a26-9711-11e7-8036-9123ffcc51e6.png">

### Desired behavior:

Run only the block with `describe.only`

### How to reproduce:

Clone repo, grab branch `double-only`, open Cypress

### Additional Info (images, stack traces, etc)

- Operating System: Mac
- Cypress Version: 0.20.0
- Browser Version: Electron

**Comment by zukoaang** [comment]
I have the same issue.
Did you ever solve this?
Hasn't anyone responded here?

**Comment by brian-mann** [comment]
This is mocha's default behavior in the version we are using. It is to be expected.

**Comment by zukoaang** [comment]
@brian-mann 
So do you mean that in the Mocha's version you're using, the `.only` just isn't supported?
That was not expected at all.
Is that intentional?
Will it be changed?

**Comment by Bkucera** [comment]
@zukoaang it seems to be a bug in mocha https://github.com/mochajs/mocha/issues/1481

**Comment by brian-mann** [comment]
We are on an older version of mocha that will match suites and tests by a regexp. When you put an `.only` on one of them - if other suites and tests match that title they will run as well.

In later versions they fixed this. We'd need to upgrade to get access to that functionality. We have deep, deep integrations into mocha so upgrading for us isn't a simple `npm install`. 

The good news is that we've recently talked about doing this so this would "just work" when we do the next upgrade.

In the mean time, simply make your titles different enough and you won't see multiple tests and suites run when using an `.only`

**Comment by jennifer-shehane** [comment]
I've verified that this will be fixed from https://github.com/cypress-io/cypress/pull/2703 - our Mocha update PR.

The code for this is done, but this has yet to be released. We'll update this issue and reference the [changelog](https://on.cypress.io/changelog) when it's released.

**Comment by jennifer-shehane** [comment]
Released in [4.0.0](https://on.cypress.io/changelog#4-0-0)
